### PR TITLE
Pad ‘^~~~^’-style diagnostic ranges properly

### PR DESCRIPTION
**Pull request by Mango0x45**
This fixes the immediate issue described in #3743.  There are other problems still though, like the fact that this will most definitely display incorrectly if the token the range covers contains non-ASCII, but this is a start!

### CODE REVIEW

File: src/error.cpp [hunk 1]
```diff
@@ -323,8 +323,13 @@ gb_internal isize show_error_on_line(TokenPos const &pos, TokenPos end) {
 		}
 		error_out("\n\t");
 
-		for (i32 i = 0; i < offset; i++) {
-			error_out(" ");
+		for (i32 rune_width, off = 0; off < offset; off += rune_width) {
+			i32 rune;
+			rune_width = utf8proc_iterate((u8 const *)line_text + off, line_len - off, &rune);
```
Comment (gingerBill) [inline src/error.cpp:328]:
```c
rune_width = cast(i32)utf8proc_iterate((u8 const *)line_text + off, line_len - off, &rune);
```

**Comment by flysand7** [comment]
I also think that if the character is a tab, the tab should be printed before ^~~^ style diagnostic. Tab width is not a fixed width. Some terminals have it configured at 8 spaces, others at 4.

**Comment by Mango0x45** [comment]
The `^~~~^` ranges should now also be printed with the proper width. Previously they would also be messed up if the error token wasn’t ASCII, but this is no longer the case.

**Comment by Kelimion** [comment]
MSVC complaints addressed. Tested on Windows and Linux with the `växjö` example.

**Comment by Kelimion** [comment]
Reverted because of regression.

**Pull request merged**